### PR TITLE
Support explain verification

### PR DESCRIPTION
**Pull request by caithagoras**
Add a new functionality to the verifier to allow explain verification.

For each query pairs to be verified, explain both control and test, and
compare the query plans. Verification is marked as succeeded as long as
the test query can be explained.

Introduced a new field matchType in the output event, which can be used
to indicate whether there is plan difference.

For non-DML queries, we don't care much about plan difference and hence
the control query and the plan comparison are skipped. 

```
== RELEASE NOTES ==

Verifier Changes
* Add support to run explain verification. (:pr:`15101`). This can be enabled by configuration property ``explain``.
```

### CODE REVIEW

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/DataVerificationConfig.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/*
```
Comment (sujay-jain) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/framework/DataVerificationConfig.java]: Curious why we are extracting this out to a separate config? Do we envision more such properties in the future?

**Comment by sujay-jain** [comment]
Still reviewing - first 2 commits LGTM

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/DataVerificationConfig.java (hunk 1 above)
Comment (caithagoras) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/framework/DataVerificationConfig.java]: I was thinking that we're only support `skip-control` for DML verification, so put this as a separate config that are specific to DML verification. But you reminded me that we could also trivially support this config for explain as well.
Removed this config and add support for `skip-control` for explain in the last commit.

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/AbstractVerification.java [hunk 2]
```diff
@@ -308,10 +303,8 @@ private VerificationResult concludeVerification(
                 sourceQuery.getName(),
                 partialResult.getStatus(),
                 partialResult.getSkippedReason(),
-                determinismAnalysis,
-                determinismAnalysis.isPresent() ?
-                        Optional.of(determinismAnalysisDetails) :
-                        Optional.empty(),
+                determinismAnalysisDetails.map(DeterminismAnalysisDetails::getAnalysis),
+                determinismAnalysisDetails,
```
Comment (sujay-jain) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/framework/AbstractVerification.java:325]: Should we change ```VerifierQueryEvent``` to just take ```determinismAnalysisDetails``` now that it contains the ```determinismAnalysis``` as well?

File: presto-verifier/src/main/java/com/facebook/presto/verifier/event/DeterminismAnalysisDetails.java [hunk 3]
```diff
@@ -28,26 +29,41 @@
...
 public class DeterminismAnalysisDetails
 {
+    private final DeterminismAnalysis analysis;
     private final List<DeterminismAnalysisRun> runs;
     private final String limitQueryAnalysis;
     private final String limitQueryAnalysisQueryId;
 
     @JsonCreator
     public DeterminismAnalysisDetails(
+            DeterminismAnalysis analysis,
```
Comment (sujay-jain) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/event/DeterminismAnalysisDetails.java:45]: nit: maybe ```determinismAnalysis```?

Comment (caithagoras) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/event/DeterminismAnalysisDetails.java:45]: done

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/AbstractVerification.java (hunk 2 above)
Comment (caithagoras) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/framework/AbstractVerification.java:325]: done

**Review by sujay-jain — APPROVED** [review]
LGTM

**Comment by caithagoras** [comment]
@mbasmanova This PR is ready now.

**Pull request merged**